database/database-value/src/lib.rs

```diff
@@ -98,7 +98,7 @@ macro_rules! as_db_bytes {
                     #[allow(clippy::size_of_in_element_count)]
                     Cow::Borrowed(slice::from_raw_parts(
                         self as *const $typ as *const u8,
-                        mem::size_of::<$typ>(),
+                        size_of::<$typ>(),
```

Comment:
What's the minimum required Rust version for this?

Reply:
Rust 1.80: https://github.com/rust-lang/rust/pull/123168

Reply:
Hm. Sounds a bit annoying to bump up our MSRV just to silence a warning. Perhaps we can simply disable the warning for now?